examples/js/loaders/GLTFLoader.js

```diff
@@ -1547,59 +1528,64 @@ THREE.GLTFLoader = ( function () {
 
 	}
 
-	function isArrayEqual( a, b ) {
+	function createGeometryKey( primitiveDef ) {
 
-		if ( a.length !== b.length ) return false;
+		var dracoExtension = primitiveDef.extensions && primitiveDef.extensions[ EXTENSIONS.KHR_DRACO_MESH_COMPRESSION ];
+		var geometryKey;
 
-		for ( var i = 0, il = a.length; i < il; i ++ ) {
+		if ( dracoExtension ) {
 
-			if ( a[ i ] !== b[ i ] ) return false;
+			geometryKey = 'draco:' + dracoExtension.bufferView
+				+ ':' + dracoExtension.indices
+				+ ':' + createAttributesKey( dracoExtension.attributes );
+
+		} else {
+
+			geometryKey = primitiveDef.indices + ':' + createAttributesKey( primitiveDef.attributes ) + primitiveDef.mode;
 
 		}
 
-		return true;
+		return geometryKey;
 
 	}
 
-	function getCachedGeometry( cache, newPrimitive ) {
+	function createAttributesKey( attributes ) {
 
-		for ( var i = 0, il = cache.length; i < il; i ++ ) {
+		var attributesKey = '';
 
-			var cached = cache[ i ];
+		for ( var key in attributes ) {
```

Comment:
I think here is assumed that keys of objects gotten by "for ( var key in object )" is consistent, correct?

My question is, is that guaranteed? It seems practically yes on the latest Chrome and FireFox, but I'm not really sure if it's guaranteed in the language spec. If no, probably we need to explicitly sort keys. Anyone knows?

Reply:
Good point, I think keys should probably be sorted yes.

Reply:
Just in case, let me confirm my understanding. glTF spec doesn't mention the order of attributes then the order can be different across primitives, correct?

https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#meshes

Reply:
The spec doesn’t (and probably couldn’t) require an order in the JSON, no.

Reply:
OK, thanks

Reply:
I updated the code to extract the keys and sort them before iterating over them